yletunnus/backends.py

```diff
@@ -10,8 +16,14 @@ class YleTunnusOAuth2(BaseOAuth2):
     REDIRECT_STATE = False
     ACCESS_TOKEN_METHOD = 'POST'
     DEFAULT_SCOPE = ['sub', 'email']
-    AUTHORIZATION_URL = 'https://auth.api.yle.fi/v1/authorize'
-    ACCESS_TOKEN_URL = 'https://auth.api.yle.fi/v1/token'
+
+    def _get_url(self, version='v1', *url_path):
+        environment = self.setting('YLETUNNUS_ENVIRONMENT')
+        if environment not in BASE_URLS.keys():
+            raise ImproperlyConfigured(
+                'Unknown YLE Tunnus environment. '
+                'Please set the settings variable SOCIAL_AUTH_YLETUNNUS_ENVIRONMENT to an allowed value.')
```

Comment:
Can the setting be checked earlier, and not during every time a URL is needed? Perhaps check the setting when this module gets imported?

Reply:
Yeah, I was thinking that we don't want to bother users who don't need this backend, but it seems that the module is loaded on-demand, so good idea.

Reply:
Very difficult to test the checking in that case, though.

Reply:
@akikoskinen I managed to make the settings checking happen on module import, but with the Tunnistamo default AUTHENTICATION_BACKENDS having yle tunnus there, Django will fail to load because the backend is loaded on startup.

@mikkokeskinen @akikoskinen a more idiomatic way might be to use https://docs.djangoproject.com/en/3.2/topics/checks/ for this. What do you think?

Reply:
Unconfigured social backends haven't prevented starting the server thus far. I wouldn't change that. I think that we should set a default value for the YLE environment and check validity like it is now.

Reply:
If the check must happen only in the case when this backend is really _used_, then the check probably has to be within some method as it is now.

Reply:
I'm still thinking we should leverage the _System check framework_ for cases where we want to do some consistency checks wrt. our configuration values (for example: YLE tunnus is enabled as a backend, has id & secret configured but no environment). But maybe that's another ticket. Let's keep this as it is.